src/main/java/com/telefonica/iot/cygnus/containers/CygnusSubscription.java

```diff
@@ -0,0 +1,371 @@
+        
+        public OrionEndpoint() {
+        } // endpoint
+        
+        public String getHost() {
+            return host;
+        } // gethost
+        
+        public String getPort() {
+            return port;
+        } // getport
+        
+        public String getSsl() {
+            return ssl;
+        } // getSsl
+        
+        public String getAuthToken() {
+            return xauthtoken;
+        } // getAuthToken
+        
+        public boolean hasAuthToken() {
+            String token = this.getAuthToken();
+            return (token != null);
+        } // hasAuthToken
+        
+        public int getEndpointSize() {
+            int size = 0;
+            size += this.getHost().length();
+            size += this.getPort().length();
+            size += this.getSsl().length();
+            if (this.hasAuthToken()) {
+                size += this.getAuthToken().length();
+            }
+            return size;
+        } // getEndpointSize
+    } // OrionEndpoint
+    
+    public class SubscriptionEntity {
+        private String type;
+        private String isPattern;
+        private String id;
+        
+        public String getEntityType() {
+            return type;
+        } // getEntityType
+        
+        public String getPattern() {
+            return isPattern;
+        } // getPattern
+        
+        public String getId() {
+            return id;
+        } // getid
+    } // SubscriptionEntity
+    
+    public class SubscriptionConditions {
+        private String type;
+        private ArrayList<String> condValues;
+        
+        public String getCondType() {
+            return type;
+        } // getType
+        
+        public ArrayList<String> getCondValues() {
+            return condValues;
+        } // getCondValues
+    } // SubscriptionConditions
+    
+    /**
+     * Checks if the given Gson has subscription and endpoint parameters.
+     * @return True if the given Json is valid as grouping rule, otherwise false
+     */
+    public int isValid() {
+        OrionSubscription orionSubscription = this.getOrionSubscription();
+        OrionEndpoint orionEndpoint = this.getOrionEndpoint();
+
+        int subscriptionMsg = isSubsciptionValid(orionSubscription);
+        int endpointMsg = isEndpointValid(orionEndpoint);
+        
+        // check if endpoint or subscription is missing
+        if ((subscriptionMsg == 1) || (endpointMsg == 1)) {
+            LOGGER.debug("Subscription or endpoint are missing.");
+            return 1;
+        } // if
+        
+        // check if subscription and endpoint contain all the required fields
+        if ((subscriptionMsg == 2) || (endpointMsg == 2)) {
+            LOGGER.debug("There are missing fields in input JSON.");
+            return 2;
+        } // if
+        
+        // check if subscription and endpoint has any empty field
+        if ((orionSubscription.getSubscriptionSize() == 0) || 
```

Comment:
Again, I think by spliting this condition we can provide detailed information about the empty fields.